consensus/src/marshal/ingress.rs

```diff
@@ -0,0 +1,204 @@
+        response: oneshot::Sender<B>,
+    },
+    /// A request to broadcast a block to all peers.
+    Broadcast {
+        /// The block to broadcast.
+        payload: B,
+    },
+    /// A notification that a block has been verified by the application.
+    Verified {
+        /// The view in which the block was verified.
+        view: u64,
+        /// The verified block.
+        payload: B,
+    },
+    /// A notarization from the consensus engine.
+    Notarization {
+        /// The notarization.
+        notarization: Notarization<V, B::Commitment>,
+    },
+    /// A finalization from the consensus engine.
+    Finalization {
+        /// The finalization.
+        finalization: Finalization<V, B::Commitment>,
+    },
+}
+
+/// A mailbox for sending messages to the marshal [`Actor`](super::actor::Actor).
+#[derive(Clone)]
+pub struct Mailbox<V: Variant, B: Block> {
+    sender: mpsc::Sender<Message<V, B>>,
+}
+
+impl<V: Variant, B: Block> Mailbox<V, B> {
+    /// Creates a new mailbox.
+    pub(super) fn new(sender: mpsc::Sender<Message<V, B>>) -> Self {
+        Self { sender }
+    }
+
+    /// Get is a best-effort attempt to retrieve a given payload from local
+    /// storage. It is not an indication to go fetch the payload from the network.
+    pub async fn get(&mut self, view: Option<u64>, payload: B::Commitment) -> oneshot::Receiver<B> {
+        let (response, receiver) = oneshot::channel();
+        if self
+            .sender
+            .send(Message::Get {
+                view,
+                payload,
+                response,
+            })
+            .await
+            .is_err()
+        {
+            error!("Failed to send get message to actor: receiver dropped");
+        }
+        receiver
+    }
+
+    /// Broadcast indicates that a payload should be sent to all peers.
+    pub async fn broadcast(&mut self, payload: B) {
+        if self
+            .sender
+            .send(Message::Broadcast { payload })
+            .await
+            .is_err()
+        {
+            error!("Failed to send broadcast message to actor: receiver dropped");
+        }
+    }
+
+    /// Notifies the actor that a block has been verified.
+    pub async fn verified(&mut self, view: u64, payload: B) {
+        if self
+            .sender
+            .send(Message::Verified { view, payload })
+            .await
+            .is_err()
+        {
+            error!("Failed to send verified message to actor: receiver dropped");
+        }
+    }
+}
+
+impl<V: Variant, B: Block> Reporter for Mailbox<V, B> {
+    type Activity = Activity<V, B::Commitment>;
+
+    async fn report(&mut self, activity: Self::Activity) {
+        let message = match activity {
+            Activity::Notarization(notarization) => Message::Notarization { notarization },
+            Activity::Finalization(finalization) => Message::Finalization { finalization },
+            _ => {
+                // Ignore other activity types
```

Comment:
A simple solution to the "pass on" functionality would just be to clone any activity that comes through here and just push it to another channel I suppose

Reply:
Fixed via https://github.com/commonwarexyz/monorepo/pull/1306/commits/b09fc266e911fa57c3034eb9b6f7a4f600711e72